app_dart/lib/src/service/datastore.dart

```diff
@@ -147,4 +151,25 @@ class DatastoreService {
       return previousStatusUpdates.single;
     }
   }
+
+  /// Generate new authorization token for [agent]
+  ///
+  /// The hashed code of token will be returned as a list
+  List<int> refreshAgentAuthToken(Agent agent){
+    const int length = 16;
+    const String urlSafeChars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+
+    final Random random = Random();
+    final List<String> token = <String>[];
+
+    for (int i = 0; i < length; i++){
+      token.add(urlSafeChars[random.nextInt(urlSafeChars.length)]);
+    }
+
+    final String hashToken = DBCrypt().hashpw(token.toString(), DBCrypt().gensalt());
+    final List<int> encodeToken = ascii.encode(hashToken);
```

Comment:
Nit: no need for the `encodeToken` local variable - you can return the value directly here.

Reply:
Updated as suggested.